query/src/main/scala/filodb/query/exec/PromQlRemoteExec.scala

```diff
@@ -136,11 +167,42 @@ case class PromQlRemoteExec(queryEndpoint: String,
         override def numRows: Option[Int] = Option(samples.size)
 
       }
-      SerializedRangeVector(rv, builder, recSchema, printTree(useNewline = false))
+      SerializedRangeVector(rv, builder, defaultRecSchema, printTree(useNewline = false))
       // TODO: Handle stitching with verbose flag
     }
-    span.finish()
-    QueryResult(id, resultSchema, rangeVectors)
+    QueryResult(id, defaultResultSchema, rangeVectors)
+  }
+
+  def genHistQueryResult(data: Data, id: String): QueryResult = {
+
+    val rangeVectors = data.result.map { r =>
+      val samples = r.values.getOrElse(Seq(r.value.get))
+
+      val rv = new RangeVector {
+        val row = new TransientHistRow()
+
+        override def key: RangeVectorKey = CustomRangeVectorKey(r.metric.map(m => m._1.utf8 -> m._2.utf8))
+
+        override def rows(): RangeVectorCursor = {
+          import NoCloseCursor._
+
+          samples.iterator.collect { case v: HistSampl =>
+            row.setLong(0, v.timestamp * 1000)
+            val sortedBucketsWithValues = v.buckets.toArray.map(x => (x._1.toDouble, x._2)).sortBy(_._1)
+            val hist = MutableHistogram(CustomBuckets(sortedBucketsWithValues.map(_._1)),
```

Comment:
You should sync with @HimaVarsha94 .... about her work to implement DDSketches, to make sure that using MutableHistograms would be correct. 

The only other reservation about this approach is that we basically need to keep all of Data in memory, which perhaps can be expensive.  However I don't have any better suggestions for now -- maybe we could serialize the incoming data into something cheaper, but that's more complicated.